tiledb/sm/subarray/subarray.h

```diff
@@ -1067,6 +1246,11 @@ class Subarray {
    */
   std::vector<RangeSetAndSuperset> range_subset_;
 
+  /**
+   * Stores LabelRangeSubset objects for handling ranges on dimension labels.
+   */
+  std::vector<LabelRangeSubset> label_range_subset_;
```

Comment:
I'm a little puzzled by this abstraction... Maybe I'm missing something, but it seems this could be cleaner as 

```
struct LabelRangeSubset {
  std::string name;
  RangeSetAndSuperSet ranges;
}
```
which could then be stored as `std::vector<optional<LabelRangeSubset>>` here, and avoid most of the definitions added earlier.

Reply:
Sounds good. Using optional + tuple directly was too verbose/confusing, but your suggestion here seems cleaner.